pkg/dashboard/adapter/adapter.go

```diff
@@ -36,31 +35,29 @@ func NewAPIService(srv *server.Server, redirector http.Handler) (*apiserver.Serv
 	}
 
 	dashboardCfg := &config.Config{
-		DataDir:    cfg.DataDir,
-		PDEndPoint: etcdCfg.ACUrls[0].String(),
-		PathPrefix: cfg.Dashboard.PublicPathPrefix,
+		DataDir:          cfg.DataDir,
+		PDEndPoint:       etcdCfg.ACUrls[0].String(),
+		PublicPathPrefix: cfg.Dashboard.PublicPathPrefix,
 	}
-	uiserver.RewriteAssetsPublicPath(cfg.Dashboard.PublicPathPrefix)
-	dashboardCfg.ClusterTLSConfig, err = cfg.Security.ToTLSConfig()
-	if err != nil {
+
+	if dashboardCfg.ClusterTLSConfig, err = cfg.Security.ToTLSConfig(); err != nil {
 		return nil, err
 	}
-	dashboardCfg.TiDBTLSConfig, err = cfg.Dashboard.ToTiDBTLSConfig()
-	if err != nil {
+	if dashboardCfg.TiDBTLSConfig, err = cfg.Dashboard.ToTiDBTLSConfig(); err != nil {
 		return nil, err
 	}
 
-	s := apiserver.NewService(
-		dashboardCfg,
-		redirector,
-		uiserver.AssetFS(),
-		func(c *config.Config, httpClient *http.Client, etcdClient *clientv3.Client) *region.PDDataProvider {
-			return &region.PDDataProvider{
-				EtcdClient:     etcdClient,
-				PeriodicGetter: input.NewCorePeriodicGetter(srv),
-			}
-		},
-	)
+	dashboardCfg.NormalizePublicPathPrefix()
 
-	return s, nil
+	return dashboardCfg, nil
+}
+
+// GenPDDataProviderConstructor generates a PDDataProviderConstructor for Dashboard API Service.
+func GenPDDataProviderConstructor(srv *server.Server) apiserver.PDDataProviderConstructor {
+	return func(c *config.Config, httpClient *http.Client, etcdClient *clientv3.Client) *region.PDDataProvider {
```

Comment:
I think it's better to add a comment about why we need `c *config.Config, httpClient *http.Client` here?

Reply:
Added a comment